src/signals/incident-management/containers/IncidentDetail/components/StatusForm/StatusForm.tsx

```diff
@@ -41,25 +45,54 @@ import type { StatusFormActions } from './actions'
 interface StatusFormProps {
   defaultTexts: DefaultTextsType
   childIncidents: IncidentChild[]
+  onClose: () => void
 }
 
-const StyledParagraph = styled.p`
-  color: themeColor('tint', 'level5');
-  margin: 0;
-`
+let lastActiveElement: HTMLElement | null = null
 
 const StatusForm: FunctionComponent<StatusFormProps> = ({
   defaultTexts,
   childIncidents,
+  onClose,
 }) => {
-  const { incident, update, close } = useContext(IncidentDetailContext)
+  const { incident, update } = useContext(IncidentDetailContext)
   const incidentAsIncident = incident as Incident
+  const { listenFor, unlisten } = useEventEmitter()
 
+  const [modalStandardTextIsOpen, setModalStandardTextIsOpen] = useState(false)
   const [state, dispatch] = useReducer<
     Reducer<State, StatusFormActions>,
     { incident: Incident; childIncidents: IncidentChild[] }
   >(reducer, { incident: incidentAsIncident, childIncidents }, init)
 
+  const openStandardTextModal = useCallback(
+    (event) => {
+      event.preventDefault()
+      disablePageScroll()
+      setModalStandardTextIsOpen(true)
+      lastActiveElement = document.activeElement as HTMLElement
+    },
+    [modalStandardTextIsOpen]
+  )
+
+  const closeStandardTextModal = useCallback(() => {
+    enablePageScroll()
+    setModalStandardTextIsOpen(false)
+
+    if (lastActiveElement) {
+      lastActiveElement.focus()
+    }
+  }, [setModalStandardTextIsOpen])
+
+  const escFunction = useCallback(
+    (event) => {
+      if (event.keyCode === 27) {
+        closeStandardTextModal()
```

Comment:
Escape closes the modal, but does it also close the status form? Like it does with the other editing forms in the `MetaList` component?

Reply:
Escape closes the modal, not the status form.